### ISSUE TITLE
Fix S3168: Rule should ignore MSTest V1 cleanup and initialize attributes

### ISSUE THREAD

**Issue by ohadschn** [SonarSource/sonar-csharp#1847]
### Description

The MSTest `ClassCleanup` method must be _"static, public, does not return a value and should not take any parameter"_ (an exception is thrown otherwise). The `void` return constraint is still in effect when the method is `async`, which triggers `S3168: "async" methods should not return "void"`.

### Repro steps

```cs
[ClassCleanup]
public static async void Cleanup() { ... }
```

### Expected behavior

Sonar should detect the `ClassCleanup` attribute and realize a `Task` could not be returned. Ideally, it would actually enforce the signature that is required by MSTest.

### Actual behavior

An error is triggered by `S3168: "async" methods should not return "void"`.

### Known workarounds

```cs
[SuppressMessage("Sonar.Bug", "S3168:AsyncMethodsShouldNotReturnVoid", Justification = "MSTest")]
```

### Related information

* SonarC# Version: 7.5.0.6605
* Visual Studio Version: 2017 (15.8.3)

**Comment by Evangelink** [comment]
Hi @ohadschn,

When doing some tests locally I have noticed that using MSTest v2 (NuGet package) you can no longer use `async void` otherwise an exception is thrown. You have to use `async Task`.

**Comment by Evangelink** [comment]
Here is my suggestion:

We will report on all cases for MSTest V2 and we will ignore `AssemblyInitialize`, `AssemblyCleanup`, `ClassInitialize`, `ClassCleanup`, `TestInitialize` and `TestCleanup` attributes for MSTest V1.

**Comment by ohadschn** [comment]
Sounds good to me 👍
